From a3dc10d1fb43aab07360aae41b460cfbc12d3d0c Mon Sep 17 00:00:00 2001
From: Zhang Yunlong <dalon.zhang@rock-chips.com>
Date: Tue, 5 Dec 2017 10:41:21 +0800
Subject: [PATCH] Revert commit "camera: rockchip: camsys_drv v0.0x22.3"

This reverts commit 42d1f377e296c6cb2c46d4c9d1e669d7448998a0.

Reason for revert: affect display when DSI uses TX1/RX1.

Change-Id: I1efeffdc21f9fcc2e53eb79b1af99431a09e23a3
Signed-off-by: Zhang Yunlong <dalon.zhang@rock-chips.com>
---
 drivers/media/video/rk_camsys/camsys_internal.h   | 4 +++-
 drivers/media/video/rk_camsys/camsys_soc_rk3288.c | 2 --
 drivers/media/video/rk_camsys/camsys_soc_rk3399.c | 2 --
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/media/video/rk_camsys/camsys_internal.h b/drivers/media/video/rk_camsys/camsys_internal.h
index ae7f5271f47c..729248e01ec8 100644
--- a/drivers/media/video/rk_camsys/camsys_internal.h
+++ b/drivers/media/video/rk_camsys/camsys_internal.h
@@ -178,9 +178,11 @@
 	1) enable SYS_STATUS_ISP status set.
 *v0.0x22.5:
 	1) gpio base start from 1000,adapt to it.
+*v0.0x22.6:
+	1) revert v0.0x22.3.
 */
 
-#define CAMSYS_DRIVER_VERSION                   KERNEL_VERSION(0, 0x22, 5)
+#define CAMSYS_DRIVER_VERSION                   KERNEL_VERSION(0, 0x22, 6)
 
 #define CAMSYS_PLATFORM_DRV_NAME                "RockChip-CamSys"
 #define CAMSYS_PLATFORM_MARVIN_NAME             "Platform_MarvinDev"
diff --git a/drivers/media/video/rk_camsys/camsys_soc_rk3288.c b/drivers/media/video/rk_camsys/camsys_soc_rk3288.c
index 91331a60b52e..e8ad148e9248 100755
--- a/drivers/media/video/rk_camsys/camsys_soc_rk3288.c
+++ b/drivers/media/video/rk_camsys/camsys_soc_rk3288.c
@@ -345,8 +345,6 @@ camsys_dev_t *camsys_dev, camsys_soc_cfg_t cfg_cmd, void *cfg_para)
 		para_int = (unsigned int *)cfg_para;
 		__raw_writel((((*para_int) & 0x03) << 3)|(0x03 << 3),
 			(void *)(camsys_dev->rk_grf_base + 0x01d4));
-		/* TX1/RX1 DPHY switch to RX status */
-		__raw_writel(0xa000a000, (void *)(camsys_dev->rk_grf_base + 0x027c));
 		break;
 	}
 
diff --git a/drivers/media/video/rk_camsys/camsys_soc_rk3399.c b/drivers/media/video/rk_camsys/camsys_soc_rk3399.c
index 749d2953913f..de14f7a759b7 100644
--- a/drivers/media/video/rk_camsys/camsys_soc_rk3399.c
+++ b/drivers/media/video/rk_camsys/camsys_soc_rk3399.c
@@ -368,8 +368,6 @@ int camsys_rk3399_cfg
 		para_int = (unsigned int *)cfg_para;
 		__raw_writel((((*para_int) & 0x03) << 3) | (0x03 << 3),
 		(void *)(camsys_dev->rk_grf_base + 0x204));
-		/* TX1/RX1 DPHY switch to RX status */
-		__raw_writel(0x300020, (void *)(camsys_dev->rk_grf_base + 0x6260));
 		break;
 	}
 
-- 
2.35.3

